tools/biomodelsML/bio_ml_tool.xml

```diff
@@ -0,0 +1,89 @@
+<tool id="bio_ml_tool" name="Random Forest model" version="0.0.1" profile="22.05">
+    <description>to predict efficacy of immune checkpoint blockade across multiple cancer patient cohorts</description>
+    <requirements>
+        <container type="docker">docker.io/anupkumar/immune-checkpoint:latest</container>
+    </requirements>
+    <command><![CDATA[
+    cp /home/\$NB_USER/forest16.onnx `pwd` && python '$biom_script'
+]]>
+    </command>
+    <configfiles>
+        <configfile name="biom_script">
+            <![CDATA[
+import os
+import argparse
+import numpy
+import onnxruntime as rt
+import pandas as pd
+
+#if $input_file.ext == "xlsx":
+data_test = pd.read_excel('$input_file')
+#elif $input_file.ext == "tabular":
+data_test = pd.read_csv('$input_file', sep="\t")
+#elif $input_file.ext == "csv":
+data_test = pd.read_csv('$input_file', sep=",")
+#end if
+
+rf16=["Cancer_Type2", "Albumin", "HED", "TMB", "FCNA", "BMI", "NLR", "Platelets", "HGB", "Stage (1:IV; 0:I-III)", "Age", "Drug (1:Combo; 0:PD1/PDL1orCTLA4)", "Chemo_before_IO (1:Yes; 0:No)","HLA_LOH", "MSI (1:Unstable; 0:Stable_Indeterminate)", "Sex (1:Male; 0:Female)"]
+x_test16 = pd.DataFrame(data_test, columns=rf16)
+sess = rt.InferenceSession(os.getcwd() + "/forest16.onnx")
+input_name = sess.get_inputs()[0].name
+label_name = sess.get_outputs()[0].name
+pred_onx = sess.run([label_name], {input_name: x_test16.astype(numpy.float32).values})[0]
+x_test16["Predictions"] = pred_onx
+x_test16.to_csv('$output_file', sep="\t", index=None)
+
+]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+	<param name="input_file" type="data" label="Test data" format="tabular,csv,xlsx" multiple="false" />
+    </inputs>
+    <outputs>
+        <data format="tabular" name="output_file" label="Predicted data"></data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_file" value="test_data.tabular" ftype="tabular" />
+            <output name="output_file" file="pred_data.tabular" compare="sim_size" delta="5" />
+        </test>
+        <test>
+            <param name="input_file" value="test_data.csv" ftype="csv" />
+            <output name="output_file" file="pred_data.tabular" compare="sim_size" delta="5" />
```

Comment:
sim_size is not recommendade, can we use assert_contents here, number of columns, number of rows https://docs.galaxyproject.org/en/latest/dev/schema.html#id112

Reply:
yes, sound good!